### PR TITLE
add user-id transformer for logs results cache

### PR DESCRIPTION
**Pull request by sandeepsukhani**
**What this PR does / why we need it**:
Adding transformer to results cache same as metrics cache done in PR #7542

### CODE REVIEW

File: pkg/querier/queryrange/log_result_cache.go [hunk 1]
```diff
@@ -105,7 +108,17 @@ func (l *logResultCache) Do(ctx context.Context, req queryrangebase.Request) (qu
...
+	transformedTenantIDs := tenantIDs
+	if l.transformer != nil {
+		transformedTenantIDs = make([]string, 0, len(tenantIDs))
+
+		for _, tenantID := range tenantIDs {
+			transformedTenantIDs = append(transformedTenantIDs, l.transformer(ctx, tenantID))
+		}
+	}
+
+	cacheKey := fmt.Sprintf("log:%s:%s:%d:%d", tenant.JoinTenantIDs(transformedTenantIDs), req.GetQuery(), interval.Nanoseconds(), alignedStart.UnixNano()/(interval.Nanoseconds()))
```
Comment (trevorwhitney) [inline pkg/querier/queryrange/log_result_cache.go:121]: I still don't fully understand the name to transform IDs either here or in the results cache. where can I find a reproduction of the bug this is causing?

**Review by trevorwhitney — COMMENTED** [review]
I don't understand the need for these transformations. I cannot reproduce a cache exploitation that results from not transforming tenant IDs.

File: pkg/querier/queryrange/log_result_cache.go (hunk 1 above)
Comment (sandeepsukhani) [reply pkg/querier/queryrange/log_result_cache.go:121]: sent you the code reproducing the issue in our internal slack thread

**Review by MichelHollands — APPROVED** [review]
LGTM

**Comment by sandeepsukhani** [comment]
drone is just getting stuck in a pending state for some image build stages. I am just going to merge the PR to get unblocked.

**Pull request merged**